cpp/example_code/transfer-manager/transferOnStream.cpp

```diff
@@ -86,35 +88,53 @@ int main(int argc, char** argv)
 
         auto uploadHandle = transfer_manager->UploadFile(LOCAL_FILE, BUCKET, KEY, "text/plain", Aws::Map<Aws::String, Aws::String>());
         uploadHandle->WaitUntilFinished();
-        std::cout << "File upload finished!" << std::endl;
-
-        // verify upload expected length of data
-        assert(uploadHandle->GetBytesTotalSize() == uploadHandle->GetBytesTransferred());
-        g_file_size = uploadHandle->GetBytesTotalSize();
-
-        // This buffer is what we used to initialize streambuf and is in memory
-        Aws::Utils::Array<unsigned char> buffer(BUFFER_SIZE);
-        auto downloadHandle = transfer_manager->DownloadFile(BUCKET,
-                KEY, 
+        bool success = uploadHandle->GetStatus() == Transfer::TransferStatus::COMPLETED; 
+      
+        if (!success)
+        {
+            auto err = uploadHandle->GetLastError();           
+            std::cout << "File upload failed:  "<< err.GetMessage() << std::endl;
+        }
+        else
+        {
+            std::cout << "File upload finished." << std::endl;
+
+            // Verify upload expected length of data
+            assert(uploadHandle->GetBytesTotalSize() == uploadHandle->GetBytesTransferred());
+            g_file_size = uploadHandle->GetBytesTotalSize();
+
+            // This buffer is what we used to initialize streambuf and is in memory
```

Comment:
Create buffer to initialize in-memory streambuf.

Reply:
Sigh, like so many of these - this junk was inherited with the example. I apparently should be going through the existing stuff much closer than I am , rather than trusting it to be quality. my bad.
here I went with "   // Create buffer to hold data received by the data stream. " instead.